libtenzir/include/tenzir/detail/weak_run_delayed.hpp

```diff
@@ -30,6 +30,24 @@ auto weak_run_delayed(caf::scheduled_actor* self, caf::timespan delay,
     caf::weak_actor_ptr{self->ctrl()});
 }
 
+/// Runs an action in a loop with a given delay without keeping the actor alive.
+///
+/// The function is first called at `start`. Even if `start` is in the past, it
+/// will be scheduled and not called immediately here.
+template <class F>
+void weak_run_delayed_loop_at(caf::scheduled_actor* self,
+                              caf::actor_clock::time_point start,
+                              caf::timespan delay, F&& f) {
```

Comment:
Would it make sense to always use this over the other `weak_run_delayed_loop` implementation to always avoid the drift? Is there a downside to it?

Reply:
That's what I did:
```cpp
template <class Function>
  requires std::is_invocable_r_v<void, std::remove_reference_t<Function>&>
void weak_run_delayed_loop(caf::scheduled_actor* self, caf::timespan delay,
                           Function&& function, bool run_immediately = true) {
  if (run_immediately) {
    std::invoke(function);
  }
  weak_run_delayed_loop_at(self, self->clock().now() + delay, delay,
                           std::forward<Function>(function));
}
```

Reply:
Okay, apparently I was just blind yesterday.